app/controllers/tasks_controller.rb

```diff
@@ -3,36 +3,64 @@ def index
   end
 
   def create
-    @list = List.find(params[:list_id])
+    @list = current_list
     @task = Task.new(task_params)
     @task.list_id = @list.id
-    @task.save
-    redirect_to list_path(@list)
+    if @task.save
+      redirect_to list_path(@list)
+    else
+      flash[:error] = 'Failed to create'
+      redirect_to new_list_task_path(@list)
```

Comment:
I think you want to do `render :new` otherwise the text they entered will be removed

Reply:
What's the difference between doing this and just not redirecting anywhere at all? @paulcsmith